code/modules/mining/laborcamp/laborstacker.dm

```diff
@@ -128,14 +129,12 @@
 
 /obj/machinery/mineral/stacking_machine/laborstacker
 	var/points = 0 //The unclaimed value of ore stacked.  Value for each ore loosely relative to its rarity.
-	var/list/ore_values = list("glass" = 1, "metal" = 2, "reinforced glass" = 4, "gold" = 20, "silver" = 20, "uranium" = 20, "titanium" = 20, "solid plasma" = 20, "plasteel" = 23, "plasma glass" = 23, "diamond" = 25, "bluespace polycrystal" = 30, "plastitanium" = 45, "bananium" = 50)
+	var/list/ore_values = list(/obj/item/stack/sheet/glass = 1, /obj/item/stack/sheet/metal = 2, /obj/item/stack/sheet/rglass = 4, /obj/item/stack/sheet/mineral/gold = 20, /obj/item/stack/sheet/mineral/silver = 20, /obj/item/stack/sheet/mineral/uranium = 20, /obj/item/stack/sheet/mineral/titanium = 20, /obj/item/stack/sheet/mineral/plasma = 20, /obj/item/stack/sheet/plasteel = 23, /obj/item/stack/sheet/plasmaglass = 23, /obj/item/stack/sheet/mineral/diamond = 25, /obj/item/stack/sheet/bluespace_crystal = 30, /obj/item/stack/sheet/mineral/plastitanium = 45, /obj/item/stack/sheet/mineral/bananium = 50)
 
 /obj/machinery/mineral/stacking_machine/laborstacker/process_sheet(obj/item/stack/sheet/inp)
-	if(istype(inp))
-		var/n = inp.name
-		var/a = inp.amount
-		if(n in ore_values)
-			points += ore_values[n] * a
+	for(var/ore in ore_values)
+		if(istype(inp, ore))
+			points += ore_values[ore] * inp.amount

```

Comment:
This is not a good long term solution, it will get slower with every ore we add.

Move ore values into the sheet type and just do `points += inp.point_value * inp.amount`

The old proc here sanity checked its input. I don't know if this is actually necessary but you may have to as well. If it isn't then this entire proc doesn't need to exist and you can just inline the above code.

Reply:
That would make generating the sheet value UI thing a bit more complex but I'll give it my best shot.